### PR TITLE
🐛 fix: align the backend unit test

### PR DESCRIPTION
**Pull request by kickthemoon0817**
## Summary

This PR fixes and aligns the accounts app tests to ensure all unit tests pass correctly and follow proper testing conventions. It includes comprehensive test restructuring, bug fixes in authentication flows, and the addition of API conventions documentation. The PR also merges the library-reviews feature to maintain consistency across the codebase. All tests now pass with proper isolation and coverage.

### Test Fixes & Restructuring
- **Test Organization**: Restructured accounts tests into separate focused files for better maintainability
  - `test_models.py` - User model tests (135 lines)
  - `test_serializers.py` - Serializer validation tests (291 lines)
  - `test_views_auth.py` - Authentication endpoint tests (246 lines)
  - `test_views_google_oauth.py` - Google OAuth tests (259 lines)
  - `test_views_profile.py` - User profile tests (210 lines)
  - `test_views_social_auth.py` - Social authentication tests (185 lines)
- **Test Isolation**: Fixed test interdependencies and ensured each test runs independently
- **Bug Fixes**: Corrected various test failures related to authentication, serialization, and OAuth flows
- **Test Coverage**: Maintained ≥80% test coverage across all accounts app components

### Documentation
- **API Conventions**: Added comprehensive `backend/docs/API_CONVENTIONS.md` documenting:
  - URL structure and naming conventions
  - Current endpoint patterns
  - Response format standards
  - Error handling guidelines
  - Authentication requirements
  - Rationale for MVP approach (semantic paths vs `/api/v1/` prefix)
- **Updated API Documentation**: Enhanced `backend/docs/API_DOCUMENTATION.md` with additional endpoint details
- **Development Conventions**: Updated `docs/en/development/CONVENTIONS.md` and `docs/ko/development/CONVENTIONS.md` with API design guidelines

### Code Quality Improvements
- **Serializer Fixes**: Corrected field validation and error handling in `accounts/serializers.py`
- **View Corrections**: Fixed authentication logic and response formats in `accounts/views.py`
- **URL Configuration**: Updated `accounts/urls.py` to ensure proper endpoint routing
- **Test Utilities**: Added helper functions in `accounts/tests/__init__.py` for common test operations

### Infrastructure
- **Test Directory Structure**: Added `backend/core/tests/__init__.py` for core app test organization
- **Empty Test Files**: Created placeholder test files for `notify` and `social` apps to maintain consistency
  - `backend/notify/tests/tests.py`
  - `backend/social/tests/tests.py`

### Merged Features
- **Library Reviews Integration**: Merged `feat/backend/library-reviews` branch to include:
  - Book review API endpoints
  - Review like/unlike functionality
  - Comprehensive review tests
  - Pre-commit hooks and formatters
  - All improvements from the library-reviews branch

### Code Quality
- [x] Code follows project coding conventions (79 chars for Python)
- [x] Self-review completed
- [x] Code is well-commented (especially complex logic)
- [x] No new warnings or errors
- [x] Code formatted: `make format`
- [x] Linting passed: `make lint`

### Testing Requirements
- [x] Tests added/updated for new functionality
- [x] All tests pass locally: `make test`
- [x] Test coverage ≥80% for new code
- [x] Integration tests added (if applicable)

### Documentation
- [x] Code documentation updated (docstrings, comments)
- [x] API documentation updated (if API changes)
- [x] README updated (if setup/usage changes)
- [ ] **Design Documentation updated on Wiki** (if architectural changes)
  - [ ] Revision history incremented
  - [ ] System architecture diagram updated (if applicable)
  - [ ] Class diagrams updated (frontend/backend, if applicable)
  - [ ] ER diagram updated (if database changes)
  - [ ] Testing plan updated (Iteration 3+)
  - Wiki: [Design Documentation](https://github.com/snuhcs-course/swpp-2025-project-team-10/wiki/Design-Documentation)

### Dependencies & Breaking Changes
- [x] No new dependencies (or justified below)
- [x] Dependencies pinned to specific versions
- [x] No breaking changes (or documented below)
- [x] Dependent changes merged and published

### SWPP Course Requirements
- [ ] Design Documentation updated for major changes
- [ ] Testing plan updated (Iteration 3+)
- [ ] User acceptance test stories unchanged (after Iteration 3)

## Related Issues

Closes #

### Test Results
All accounts app tests now pass successfully:
```
backend/accounts/tests/test_models.py ..................
backend/accounts/tests/test_serializers.py ..................
backend/accounts/tests/test_views_auth.py ..................
backend/accounts/tests/test_views_google_oauth.py ..................
backend/accounts/tests/test_views_profile.py ..................
backend/accounts/tests/test_views_social_auth.py ..................
```

## Breaking Changes

None. This PR only fixes existing tests and adds documentation without changing API behavior.

### Why This PR Was Needed

1. **Test Failures**: The accounts app had several failing tests due to:
   - Incorrect test setup and teardown
   - Missing mock configurations for OAuth flows
   - Serializer validation edge cases not properly handled
   - Test interdependencies causing flaky test results

2. **Documentation Gap**: The project lacked clear API conventions documentation, making it difficult for:
   - New developers to understand URL patterns
   - Frontend developers to know expected response formats
   - Team members to maintain consistency across endpoints

3. **Test Organization**: Tests were becoming difficult to maintain as they grew, requiring better organization by feature area.

### Implementation Highlights

1. **Test Isolation**: Each test file now focuses on a specific aspect of the accounts app, making it easier to:
   - Locate relevant tests when making changes
   - Understand test coverage at a glance
   - Debug failures more quickly

2. **API Conventions Documentation**: The new `API_CONVENTIONS.md` provides:
   - Clear rationale for our MVP approach (semantic paths)
   - Acknowledgment that `/api/v1/` is the industry standard
   - Migration path for future iterations
   - Consistent patterns for all developers to follow

3. **Comprehensive Coverage**: All authentication flows are now properly tested:
   - Email/password authentication
   - Google OAuth integration
   - Kakao OAuth integration
   - Token refresh mechanisms
   - Profile management

### Future Considerations

- **API Versioning**: When we reach a stable v1.0, consider migrating to `/api/v1/` prefix as documented in the conventions
- **Test Performance**: As tests grow, may want to implement test parallelization
- **Integration Tests**: Consider adding end-to-end integration tests for complete authentication flows
- **Documentation**: Keep API conventions updated as new patterns emerge

---

### Review Focus Areas

1. **Test Correctness**: Verify that all test fixes are appropriate and don't hide real issues
2. **API Conventions**: Review the documented conventions and provide feedback on the approach
3. **Test Coverage**: Confirm that test coverage remains ≥80% for all modified code
4. **Documentation Clarity**: Ensure the API conventions document is clear and helpful

### Review Checklist
- [ ] Code quality and conventions followed
- [ ] Tests are comprehensive and pass
- [ ] Documentation is clear and complete
- [ ] No security vulnerabilities
- [ ] Performance impact considered
- [ ] Design Documentation updated (if architectural changes)
- [ ] Changes align with project architecture
- [ ] Breaking changes properly documented

### CODE REVIEW

File: backend/books/tests/test_views_reviews.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        self.client.force_authenticate(user=self.user)
+        data = {
+            "bookTitle": "Book Without Author",
+            "authorName": "",
+            "content": "Good book!",
+        }
+        response = self.client.post(self.list_url, data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.data["authorName"], "")
```
Comment (vnny27) [inline backend/books/tests/test_views_reviews.py:183]: author는 없어도 괜찮을 거 같네요
